### ISSUE TITLE
JS debugger Network tab not showing fetch requests

### ISSUE THREAD

**Issue by maersk96** [expo/expo#23633]
### Summary

I just upgraded my project to the Expo SDK 49. I cannot get the JS debugger to work. I am using Hermes, a SDK version 49.0.0 dev-client app. When opening the JS debugger (through the dev menu or by pressing J) I can correctly see logs from the app, so the connection to the device should be alright. When navigating to the Network tab nothing shows up on fetch requests. What could be the cause for this? Am i missing something?

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

iOS

### Package versions

expo-dev-client: 2.4.6
expo: 49.0.3

Google Chrome: Version 114.0.5735.198 (Official Build) (arm64)

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.0
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.12.1 - /usr/local/bin/node
      npm: 9.7.2 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.12.1 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 3.16.0
    Expo Workflow: managed

### Reproducible demo

Hard to reproduce.

**Comment by vendeza** [comment]
same

**Comment by iway1** [comment]
this happens all the time. sometimes it works, sometimes it doesn't. it's really frustrating (not to mention a waste of the developer's time)

**Comment by davevilela** [comment]
Any updates? Why was this issue closed?

**Comment by ankitjasbeersingh** [comment]
I facing same issue

**Comment by GrinZero** [comment]
same

**Comment by GrinZero** [comment]
Everyone can try this solution, and I have successfully made the network tab work again:
1. Run 'pnpm ios' or other startup commands while opening devtool, but the network tab will not work at this time.
2. Turn off the service that runs the startup command, but do not turn off devtool. The devtool is in a waiting to reconnect state.
3. Rerun 'pnpm ios' or other startup commands, while continuously clicking on' reconnect 'on devtool until reconnected.

After the previous connection, The network tab can work continuously, and I speculate it is related to caching. The cause of the problem may be related to devtool startup The timing of server attempts to connect is related.

**Comment by rhyek** [comment]
seeing this as well. new to react native/expo in general and it is surprising to me this hasn't gained more attention. @GrinZero's recommendation works, but not sure how reliable it will be.

**Comment by samimenouar** [comment]
I'm surprised too, I have this problem and @GrinZero's fix works for me

**Comment by vlanemcev** [comment]
The same problem here

**Comment by shahidul5** [comment]
any update? I'm facing the same problem here. I'm using - 

expo 50.0.13
react-native 0.73.4

**Comment by GabrielDierks** [comment]
Omg @GrinZero it actually works <3 . But its quite crazy that you are forced to do this. Expo team PLEASE look into this. cc @byCedric

**Comment by kurucaner** [comment]
I've found a workaround that might help:

1. Open your application in the simulator.
2. Open the JS Debugger.
3. Close your application in the simulator.
4. You'll see a "WebSocket disconnected" error—keep it open.
5. Reopen your application and wait until it fully loads. (I've encountered errors when reconnecting during the splash screen.)
6. Once the app is fully open, click "Reconnect DevTools."
7. Go to the Networks tab, and you should see the results.

**Comment by grahamseamans** [comment]
Switching to the `npx expo start` as my launch command fixed this for me.

**Comment by Moakeson** [comment]
I'm currently having issues with this as well. @grahamseamans suggestion worked once for me but I haven't been able to make it work again yet.

I'm using axios for handling the network request.

I'd really hope that there can be a fix provided for this in the near future.

expo 51.0.28
react 0.74.5

**Comment by alimukhtarcc** [comment]
**1.** Open the Debugger:
- Ensure your device or emulator is connected to server.
- If not, press ```j``` to open the debugger.

**2.** Reload the App if Origin is Not Visible:

- If you cannot see the **Origin**, reload your app from the Expo menu.

**3.** Navigate to Security tab:
- Once the **Origin** appears under the **Security** section, click on **"View Requests in Network panel"**  to inspect all network activities.
![image](https://github.com/user-attachments/assets/4b4ab4b5-d0b1-4d1c-b7fb-1e57a57178e3)


**4.** Manually Navigate to the Network Panel (If Necessary):
- If clicking "View Requests" does not automatically take you to the Network panel, manually enter your domain and scheme in the debugger:
![image](https://github.com/user-attachments/assets/3642e530-167d-4185-ac6a-f19ec9e2702b)

**Comment by mdecrocemovson** [comment]
I am still getting crushed here, seems like it works every once in a while. Very confusing

**Comment by padoski** [comment]
not working still

**Comment by samwyness** [comment]
Facing the same issue. I'm running expo 52.0.14 and react-native 0.76.3

**Comment by fatihkayan20** [comment]
I can get requests after installing "expo-dev-client" package

**Comment by nicholashamilton** [comment]
Not working, so much wasted time
expo 52.0.19
react 0.76.5

**Comment by nicholashamilton** [comment]
I spent a good amount of time trying to get the network tab to work with `expo start` on a simulator development build, no luck.

What fixed the issue for me was running `expo run:ios`!

expo 52.0.19
react 0.76.5

I'm using a Nx mono repo so the command for me was `npx nx run-ios app`.

**Comment by luffy-taro** [comment]
With axios, I had an error in the request interceptor with no error thrown to console, in the response interceptor the error response would log an empty object. 

Fixed the bug in the request interceptor and network logs now show up.

**Comment by fatihkayan20** [comment]
> I can get requests after installing "expo-dev-client" package

Copy as curl and other copy options not working

**Comment by arthwood** [comment]
Upgrading `expo-dev-client` from `4.0.20` to `5.0.6` helped here.

**Comment by robertwt7** [comment]
mine also never works on windows and android

**Comment by peterchaula** [comment]
A lot of devs coming from might be used to filtering axios requests using the "Fetch/XHR" tab. But this doesn't work in the expo network inspector. All requests are logged under the "All" tab.

**Comment by azhaochen** [comment]
Still seeing this issue
```
"expo": "~52.0.27",
"expo-dev-client": "~5.0.10",
"react-native": "0.76.6",
```

**Comment by pedrollcopatti** [comment]
For some reason, the only way that works for me is launching with `npx expo start` and clicking `j` in the terminal. But it is not guaranteed.

**Comment by sirus-max** [comment]
Struggled a lot but now I have started using **Charles Proxy** to debug network requests. It requires some initial setup like installing certificates on simulators and on your computer. But once set up, it works every time as expected.

**Comment by rgomezp** [comment]
`npx expo run ios` worked for me. Kind of annoying though because it rebuilds the app

**Comment by mly-johndpope** [comment]
may or may not work - your mileage may vary. (before i clicked this - network requests were empty even though it was selected - beeen troubleshooting this for days so was surprised and delighted that this worked.) im on m1 / expo 53 / react native .79 / react 19 - Google chrome version  137.0.7151.55  <img width="681" alt="Image" src="https://github.com/user-attachments/assets/ef65c931-6a34-40e8-b32d-42579ca406eb" />

**Comment by 0x1337ak** [comment]
~~After trying to set up [Reactotron](https://docs.infinite.red/reactotron/quick-start/react-native/) which I didn’t like because of its UI, I realized their setup might also work for React Native DevTools, and it did.~~
~~expo 53~~
~~"react-native": "0.79.5",~~
nvm

**Comment by leopego** [comment]
I started a new project, but it didn't come with expo-dev-client, I just installed and it worked!

**Comment by ItanSutarlan** [comment]
installing expo-dev-client helps me to solve the issue not able to inspect the network request:
https://docs.expo.dev/versions/latest/sdk/dev-client/#configuration-in-app-config

**Comment by LovesWorking** [comment]
This still happens to me! If you can't get it to work just use the internal dev tools https://github.com/LovesWorking/react-native-buoy

**Comment by 0x1337ak** [comment]
**I found this workaround when the network inspector stops working:**

1. Start Metro.
2. Completely close the app on the simulator (force-quit it).
3. Reopen the app on the simulator.
4. Press **`j`** in your terminal to open the network inspector or hit **Reconnect** if `j` doesn’t work (keep the inspector open).

**Comment by drnow4u** [comment]
> **I found this workaround when the network inspector stops working:**
> 
> 1. Start Metro.
> 2. Completely close the app on the simulator (force-quit it).
> 3. Reopen the app on the simulator.
> 4. Press **`j`** in your terminal to open the network inspector or hit **Reconnect** if `j` doesn’t work (keep the inspector open).

Works for me 👍

**Comment by PrabhatDongare** [comment]
@radko93, @vonovak, @kitten, @suveshmoza, @alanjhughes, @amandeepmittal 
Hi Guys I tagged some of the contributors in a hope this issue will be resolved. I'm still facing the same issue where the network tab is not showing anything. I've tried all the solutions mentioned above other than moving to development build and an in app debugger solution package @LovesWorking. Can anyone look into it and see if we can have some fixes.
